docs/design/modular_installation.md

```diff
@@ -0,0 +1,63 @@
+# Kuadrant Proposal - Modular Installation
+
+Kuadrant is developing a set of loosely coupled functionalities built directly on top of Kubernetes.
+Kuadrant aims to allow customers to just install, use and understand those functionalities they need.
+Currently, the installation tool of kuadrant, the [kuadrantctl CLI](https://github.com/Kuadrant/kuadrantctl),
+installs all or nothing. For instance, a customer not wanting to rate limit the API,
+should not have to install the rate limit component ([Limitador](https://github.com/Kuadrant/limitador)).
+
+## High Level Goals
+
+* Install only required components. Operate only required components.
+* Expose only the kuadrant API that has available implementation by the with the deployed components
+
+## Installation profiles
+
+The kuadrant installation mechanism should offer modular installation to enable pieces of kuadrant.
+Modular installation options should be feature oriented rather than deployment component oriented.
+Then, it is up to the installation tool to decide what components need to be deployed and how to
+configure it.
+
+Kuadrant features:
+
+* Multiple APIs bundled in a product
+* Service Discovery
+* TLS
+* Authentication (mTLS, API key, OIDC)
+* Rate Limiting (pre-auth, post-auth)
+
+Each feature, or part of it, is eligible to be included or excluded when installing kuadrant.
+
+Some profiles can be defined to group set of commonly required features. Naming the profiles
+allows the customer to easily express wanted installation configuration. Furthermore, profiles
+not only can be used to group a set of features, profiles can be used to define deployment options.
+
+| Name | Description |
+| ---  | --- |
+| **Minimal** | Minimal installation required to run an API without any protection, analytics or API management |
+| **Authenticated** | MTLS, API key and OIDC authentication features |
+| **RateLimit** | Basic rate limit (only pre-auth rate limit) features |
+| **Full** | Full featured kuadrant installation |
+| **Development** | Full featured kuadrant for development purposes. Development deployment mode: debug log level, no cache, tracing |
+
+A kuadrant operator, together with a design of a kuadrant CRD is desired.
```

Comment:
Do we want to propose a CRD design here or elsewhere?

Reply:
Not here :)

Reply:
After re-reading this, I guess I answered too quickly thinking about kuadrant configuration objects (APiProduct, APIs, AuthConfigs, RateLimits)... but maybe we can add some first draft for the `kuadrant CRD`.  @mikenairn do you have already something in mind? 

Something very basic would be:

```yaml
apiVersion: install.kuadrant.io/v1alpha1
kind: Kuadrant
metadata:
  namespace: kuadrant-system
  name: kuadrantcontrolplane
spec:
  profile: full
  components:
     authorino: {}
     limitador: {}
```
I wonder if we need to add some customization of istio. Latest discussions held in other channels indicated that istio would not be owned by kuadrant and should be a given in the cluster, so customization of istio does not make sense. Adding a custom and additional istio gateway would be an option

Reply:
This is an  interesting point of discussion. From a getting started point of view, it may make sense to have the option to install istio for now I think lets point ppl at the istio docs to do  that. As you say taking a gateway name and namespace or perhaps label selector would be needed. Then we either create it if doesn't exist use it if it does exist (for current impl we create our envoyfilters etc and apply to this gateway workload).